verilog/analysis/verilog_filelist.cc

```diff
@@ -27,24 +27,13 @@
 #include "common/util/iterator_range.h"
 
 namespace verilog {
-void FileList::Append(const FileList& other) {
-  file_paths.insert(file_paths.end(), other.file_paths.begin(),
-                    other.file_paths.end());
-  preprocessing.include_dirs.insert(preprocessing.include_dirs.end(),
-                                    other.preprocessing.include_dirs.begin(),
-                                    other.preprocessing.include_dirs.end());
-  preprocessing.defines.insert(preprocessing.defines.end(),
-                               other.preprocessing.defines.begin(),
-                               other.preprocessing.defines.end());
-}
-
-FileList ParseSourceFileList(absl::string_view file_list_path,
-                             const std::string& file_list_content) {
+absl::Status AppendFileListFromContent(absl::string_view file_list_path,
+                                       const std::string& file_list_content,
+                                       FileList* append_to) {
   // TODO(hzeller): parse +define+ and stash into preprocessing configuration.
   constexpr absl::string_view kIncludeDirPrefix = "+incdir+";
-  FileList file_list_out;
-  file_list_out.file_list_path = std::string(file_list_path);
-  file_list_out.preprocessing.include_dirs.push_back(".");
+  append_to->file_list_path = std::string(file_list_path);
+  append_to->preprocessing.include_dirs.push_back(".");  // Should we do that?
```

Comment:
Not entirely sure if the standard has an opinion about this, or if it is implementation defined.

Reply:
At least for the accumulation use-case, it would then probably be something that would be done once at the toplevel.